packages/contracts-bedrock/contracts/L1/SystemConfig.sol

```diff
@@ -207,4 +257,69 @@ contract SystemConfig is OwnableUpgradeable, Semver {
             sstore(slot, _unsafeBlockSigner)
         }
     }
+
+    /**
+     * @notice A getter for the resource config. Ensures that the struct is
+     *         returned instead of a tuple.
+     */
+    function resourceConfig() external view returns (ResourceConfig memory) {
+        return _resourceConfig;
+    }
+
+    /**
+     * @notice An external setter for the resource config. In the future, this
+     *         method may emit an event that the `op-node` picks up for when the
+     *         resource config is changed.
+     *
+     * @param _config The new resource config values.
+     */
+    function setResourceConfig(ResourceConfig memory _config) external onlyOwner {
+        _setResourceConfig(_config);
+    }
+
+    /**
+     * @notice An internal setter for the resource config. Ensures that the
+     *         config is sane before storing it. Holds the following invariants:
+     *         - min base fee must be less than or equal to max base fee
+     *         - base fee change denominator must be greater than 0
+     *         - max resource limit plus system tx gas must be less than or
+     *           equal to the L2 gas limit
+     *         - elasticity multiplier must be greater than 0
+     *         - no precision loss when computing target resource limit
+     *
+     * @param _config The new resource config
+     */
+    function _setResourceConfig(ResourceConfig memory _config) internal {
+        require(
```

Comment:
Might be slightly cleaner if the invariants are explained inline with each of the require statements, rather than as the top-level function comment.